cunumeric/linalg/linalg.py

```diff
@@ -401,43 +429,48 @@ def norm(x, ord=None, axis=None, keepdims=False):
     # Normalize the `axis` argument to a tuple.
     nd = x.ndim
     if axis is None:
-        axis = tuple(range(nd))
+        computed_axis = tuple(range(nd))
+    elif not isinstance(axis, tuple):
+        computed_axis = (axis,)
     else:
-        if not isinstance(axis, tuple):
-            axis = (axis,)
-        for ax in axis:
-            if not isinstance(ax, int):
-                raise TypeError(
-                    "`axis` must be None, an integer or a tuple of integers"
-                )
-
-    if len(axis) == 1:
+        computed_axis = axis
+    for ax in computed_axis:
+        if not isinstance(ax, int):
+            raise TypeError(
+                "`axis` must be None, an integer or a tuple of integers"
+            )
```

Comment:
The very slight difference in this logic is that the check will also happen for the `axis=None` case, which is slightly redundant since it will always succeed in that case by design. I think reducing the nested complexity is worth the trade-off tho.